### ISSUE TITLE
RN 0.72 new arch

### ISSUE THREAD

**Issue by billnbell** [react-native-cameraroll/react-native-cameraroll#511]
Getting:

UnsupportedGenericParserError: Module NativeCameraRollModule: Unrecognized generic type 'SubTypes' in NativeModule spec.

```
[Codegen] >>>>> Processing rncameraroll


[Codegen] Done.
/Users/bill_1/theclub/TheClub/node_modules/flow-parser/flow_parser.js:818
throw a}function
^

UnsupportedGenericParserError: Module NativeCameraRollModule: Unrecognized generic type 'SubTypes' in NativeModule spec.
    at translateTypeReferenceAnnotation (/Users/bill_1/theclub/TheClub/node_modules/@react-native/codegen/lib/parsers/typescript/modules/index.js:153:15)
    at translateTypeAnnotation (/Users/bill_1/theclub/TheClub/node_modules/@react-native/codegen/lib/parsers/typescript/modules/index.js:224:14)
    at parseObjectProperty (/Users/bill_1/theclub/TheClub/node_modules/@react-native/codegen/lib/parsers/parsers-commons.js:261:7)
    at /Users/bill_1/theclub/TheClub/node_modules/@react-native/codegen/lib/parsers/typescript/modules/index.js:61:16
    at guard (/Users/bill_1/theclub/TheClub/node_modules/@react-native/codegen/lib/parsers/utils.js:26:14)
    at /Users/bill_1/theclub/TheClub/node_modules/@react-native/codegen/lib/parsers/typescript/modules/index.js:60:14
    at Array.map (<anonymous>)
    at translateObjectTypeAnnotation (/Users/bill_1/theclub/TheClub/node_modules/@react-native/codegen/lib/parsers/typescript/modules/index.js:59:6)
    at translateTypeAnnotation (/Users/bill_1/theclub/TheClub/node_modules/@react-native/codegen/lib/parsers/typescript/modules/index.js:320:14)
    at parseObjectProperty (/Users/bill_1/theclub/TheClub/node_modules/@react-native/codegen/lib/parsers/parsers-commons.js:261:7) {
  nodes: [
    Node {
      type: 'TSTypeReference',
      start: 414,
      end: 422,
      loc: SourceLocation {
        start: Position { line: 16, column: 14, index: 414 },
        end: Position { line: 16, column: 22, index: 422 },
        filename: undefined,
        identifierName: undefined
      },
      typeName: Node {
        type: 'Identifier',
        start: 414,
        end: 422,
        loc: SourceLocation {
          start: Position { line: 16, column: 14, index: 414 },
          end: Position { line: 16, column: 22, index: 422 },
          filename: undefined,
          identifierName: 'SubTypes'
        },
        name: 'SubTypes'
      }
    }
  ]
}
```

**Comment by Arjan-Zuidema** [comment]
Have you seen [this](https://github.com/react-native-cameraroll/react-native-cameraroll/issues/498) issue?

**Comment by billnbell** [comment]
I got it to work in reverse way...

unsure if this makes any sense ... @react-native-camera-roll+camera-roll+5.7.1.patch
```
node_modules/@react-native-camera-roll/camera-roll/src/NativeCameraRollModule.ts
```

And

```
diff --git a/node_modules/@react-native-camera-roll/camera-roll/src/NativeCameraRollModule.ts b/node_modules/@react-native-camera-roll/camera-roll/src/NativeCameraRollModule.ts
index 7bcf589..4904762 100644
--- a/node_modules/@react-native-camera-roll/camera-roll/src/NativeCameraRollModule.ts
+++ b/node_modules/@react-native-camera-roll/camera-roll/src/NativeCameraRollModule.ts
@@ -13,7 +13,7 @@ type Album = {
 type PhotoIdentifier = {
   node: {
     type: string;
-    subTypes: SubTypes;
+//    subTypes: SubTypes;
     group_name: string;
     image: {
       filename: string | null;
@@ -57,6 +57,7 @@ export interface Spec extends TurboModule {
     internalID: string,
     options: Object,
   ): Promise<PhotoIdentifier>;
+  // Subtypes: SubTypes;
 }
 
 export default TurboModuleRegistry.getEnforcing<Spec>('RNCCameraRoll');
```

**Comment by billnbell** [comment]
OK I can add it back but not thru the type.

```
    subTypes:
      | 'PhotoPanorama'
      | 'PhotoHDR'
      | 'PhotoScreenshot'
      | 'PhotoLive'
      | 'PhotoDepthEffect'
      | 'VideoStreamed'
      | 'VideoHighFrameRate'
      | 'VideoTimelapse';
```

**Comment by billnbell** [comment]
OK - From another thread for codegen.
```
If the type is only used in a single native module/component, exporting the type from that specific spec file is mostly fine. But as soon as the type is used across multiple native modules/components, the type will have to be duplicated in all the specs using it.
```
https://github.com/reactwg/react-native-new-architecture/discussions/91

[@Simon-Techform](https://github.com/Simon-TechForm)

you can bring in the SubTypes and it works too.

```
type SubTypes =
  | 'PhotoPanorama'
  | 'PhotoHDR'
  | 'PhotoScreenshot'
  | 'PhotoLive'
  | 'PhotoDepthEffect'
  | 'VideoStreamed'
  | 'VideoHighFrameRate'
  | 'VideoTimelapse';
```

**Comment by Arjan-Zuidema** [comment]
Should be fixed in 7.5.2
